### ISSUE TITLE
add submodule support

### ISSUE THREAD

**Issue by rejon** [mssun/passforios#77]
I have my personal passwords and then want to store my project password in a separate folder for access control issues. 

Can your software just do the next step to do:

git submodule init
git submodule update

?

Thanks
Jon

**Comment by mssun** [comment]
Hi Jon, thanks for the suggestion. I understand your idea, but we should support multiple keys in the first place. Since submodule may have different keys, we should provide matching keys for encryption/decryption.

**Comment by rejon** [comment]
yeah you mean first?

**Comment by mssun** [comment]
Yes, we will figure how to support multiple pairs of keys first.

**Comment by rejon** [comment]
ok makes sense,  but for default, if no key in the submodule root dir then should default to the gpg key in the directory above, correct?

**Comment by yishilin14** [comment]
I guess we should find `.gpg-id` in the following order: current dir, parent dir ,..., root. And use the first `.gpg-id` found. Correct?

We will play around `git submodule` in Pass (command line) first, and then implement it in Pass for iOS.

**Comment by rejon** [comment]
sounds like a plan

**Comment by yishilin14** [comment]
It seems like we could solve issue #52 together with this one.

- Both features require importing multiple keys (especially multiple public keys).
- Both features require encrypting files using public keys indicated in the "nearest" `.gpg-id` config.

Just my two cents.

**Comment by lukasbestle** [comment]
What is the state of this feature? As far as I can tell, `.gpg-id` is already supported now. But I think it isn't a requirement as there are also use cases where a submodule may make sense for a single user. And even when multiple users are involved, read-only access works in any case.

I think the feature should work like this:

- By default, the app should run `git submodule update --init --recursive` on every pull.
- There could be a switch in the options that enables the `--remote` flag (so the full command would be `git submodule update --init --recursive --remote`). This ensures that the Git submodule is updated to its latest commit, not the one that's referenced in the main repository.
